README.md

```diff
@@ -7,11 +7,139 @@ maintaining rather than effacing provenance.
+performing an outer right join, and synthesizing accessibility data suitable for visualisation with
+[https://github.com/inclusive-design/covid-data-monitor].
+
+### The tangled mat
+
+The basic operation of the tangled mat is illustrated in diagram [Tangled Mat](https://docs.google.com/drawings/d/1OIT6zN0jwwuyt4ZmFeA-eWJiFEert8uM2zLtReqkve0/edit).
+Several layers of arbitrary JSON shape may be combined, with a resulting structure which is the result of a deep merge
+as if they had been combined using
+
+    jQuery/fluid.merge(true, []/{}, layer1, layer2 ...)
+
+With the difference that
+
+1. The merging is performed in a lazy manner - a result in the merged structure is only evaluated at the point it is
+demanded,
+2. The provenance of each final object in the merged structure is available in a "provenance map" which is isomorphic
+to the merged value, recording which layer each resulting leaf value was drawn from
+
+The primary API of the mat consists of its creator function `fluid.tangledMat`, and methods `addLayer`, `getRoot`,
+`readMember`, `getProvenance`, `evaluateFully` and `setWritableLayer` - see JSDocs. An example session interacting with the
+mat mirroring the example drawing:
+
+````javascript
+var mat = fluid.tangledMat([{
+    value: {
+        a: 1
+    },
+    name: "Layer1"
+}, {
+    value: {
+        b: 2
+    },
+    name: "Layer2"
+}]);
+
+mat.addLayer(
+    {
+        c: {
+            d: 4
+        }
+    }, "Layer3"
+);
+
+var provenanceMap = mat.getProvenance();
+// returns {
+//     a: "Layer1",
+//     b: "Layer2",
+//     c: {
+//        d: "Layer3"
+//     }
+// }
+````
+
+### The data pipeline
+
+The pipeline is configured by a JSON5 structure with top-level elements
+
+````text
+    type {String} The name of the overall pipeline (will eventually become a grade name of some sort) 
+     
+    datasets: Free hash of dataset structures each with elements
+        sourceType: "git", (currently the only supported type)
+        repository: <git repository URL, e.g. "https://github.com/inclusive-design/covid-assessment-centres">
+        path: <String path within the repository, e.g. "WeCount/assessment_centre_data_collection_2020_09_02.csv">
+    },
+
+    pipeline: Free hash of pipeline elements with elements
+        type: <String - global function name>
+        <other type-dependent properties>
+````
+
+Pipeline elements accept and return records as the following triple:
+
+````text
+    {
+        value: <Array of CSV row values, as loaded from the dataset, with each row as a hash>
+        provenance: <Isomorphic to value, with a provenance string for each data value - as per tangled mat's provenance> 
+        provenanceMap: <A map of provenance strings to records resolving the provenance - either a dataset record or another pipeline record>
+    }
+````
+
+Pipeline elements available include
+
+    fluid.forgivingJoin - executes an inner or outer join given two CSV structures
```

Comment:
This pipeline seems supporting inner or outer right join, inner or outer left join. Can you explain what they are or add a link to the explanations? Thanks.

Reply:
Added